.github/workflows/after-pullrequest.yml

```diff
@@ -15,21 +15,26 @@ jobs:
   report-test-results:
     name: Report Test Results
     runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion != 'skipped'
+
+    permissions:
+      checks: write
+      pull-requests: write
+      actions: read
+      # only needed for private repository
+      # contents: read
+      # issues: read
     steps:
-      - name: Upload Unit Test Results
-        uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226 #v1.6.0
+      - name: Download and Extract Artifacts
+        uses: dawidd6/action-download-artifact@71072fbb1229e1317f1a8de6b04206afb461bd67 # v3.1.2"
         with:
-          artifact: unit-tests
-          name: Unit Test Results
-          path: "**/*.trx"
-          reporter: dotnet-trx
-          fail-on-error: false
+          run_id: ${{ github.event.workflow_run.id }}
+          path: artifacts
 
-      - name: Upload E2E Test Results
-        uses: dorny/test-reporter@c9b3d0e2bd2a4e96aaf424dbaa31c46b42318226 #v1.6.0
+      - name: Publish Test Results
```

Comment:
I _think_ this will work for both, though not 100% sure until we merge something.

It looks like all artifacts are retrieved and unzipped into `artifacts`, so `**/*.trx` would catch both files.